pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/jdbc/ThinPreparedStatement.java

```diff
@@ -201,9 +200,10 @@ public ResultSet executeQuery( IDataServiceClientService.StreamingMode windowMod
   }
 
   @Override
-  public Observable<List<RowMetaAndData>> executePushQuery( IStreamingParams streamParams )
-    throws Exception {
-    return super.executePushQuery( replaceSql(), streamParams );
+  public void executePushQuery( IStreamingParams streamParams,
+                                Observer<List<RowMetaAndData>> consumer,
+                                AtomicBoolean consumerKeepAlive ) throws Exception {
```

Comment:
we shouldn't be using an AtomicBoolean, the Observer should call dispose() after subscribe to cancel the connection

Reply:
The issue with that approach is that the consumer is subscribed on the dataservices server, but the disposable is needed on the client/browser side, since that's where the dispose would be called.

It may be possible to return that disposable as a return for the method call, but why is that better than having a simple atomic boolean being used to control that from the client side?